presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java

```diff
@@ -938,7 +939,12 @@ public SystemSessionProperties(
                         SPARK_ASSIGN_BUCKET_TO_PARTITION_FOR_PARTITIONED_TABLE_WRITE_ENABLED,
                         "Assign bucket to partition map for partitioned table write when adding an exchange",
                         featuresConfig.isPrestoSparkAssignBucketToPartitionForPartitionedTableWriteEnabled(),
-                        true));
+                        true),
+                booleanProperty(
+                        LOG_FORMATTED_QUERY_ENABLED,
+                        "Log formatted prepared query instead of raw query when enabled",
+                        featuresConfig.isLogFormattedQueryEnabled(),
+                        false));
```

Comment:
Do we need a config to guard?

Reply:
I think having a config is better because there might be external dependencies that relies on the old behavior, and this change will change the query string exposed via `QueryCreatedEvent` and `QueryCompletionEvent`.